app/components/publications/publication/proofs/proof-edit-modal.js

```diff
@@ -0,0 +1,91 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import {
+  isNone, isPresent
+} from '@ember/utils';
+import { tracked } from '@glimmer/tracking';
+import { task } from 'ember-concurrency-decorators';
+import {
+  ValidatorSet, Validator
+} from 'frontend-kaleidos/utils/validators';
+
+/**
+ * @argument {PublicationFlow} publicationFlow
+ * @argument {Piece} piece
+ * @argument onSave
+ * @argument onCancel
+ */
+export default class PublicationsPublicationProofsProofEditModalComponent extends Component {
+  @service currentSession;
+
+  @tracked file = this.args.piece.file;
```

Comment:
Deze relatie is ook async en kan beter bij in `loadData` gezet worden.

Reply:
Eigenlijk kon die weg, gezien die hier enkel in de template nodig is.

i.v.m. de async in het algemeen: Deze heb ik kunnen vermijden door de if condities in de check voor receivedAtDate te zetten. Deze is nu wel moeilijk geworden.
Ik dacht dat het nuttig was om de receivedAtDate te resetten als deze door de gebruiker gecleard werd zonder dat een nieuwe date werd ingevoerd (het was dan waarschijnlijk een vergissing) Eigenlijk kan de validatie van receivedAtDate ook weg dan.